enterprise/server/remote_execution/dirtools/dirtools.go

```diff
@@ -558,7 +565,7 @@ func (ff *BatchFileFetcher) FetchFiles(filesToFetch FileMap, opts *DownloadTreeO
 		// fit in the batch call, so we'll have to bytestream
 		// it.
 		size := d.GetSizeBytes()
-		if size > gRPCMaxSize {
+		if size > gRPCMaxSize || ff.casClient == nil {
```

Comment:
Is there a reason to disable batching for this use case? If not, maybe just pass a CAS client to the BatchFileFetcher to avoid adding more conditional logic

Reply:
In v0 for firecracker, I only implemented a ByteStream proxy because 1) reads from the guest generally can't be batched and 2) for uploads, it's not clear how much perf benefit there is to using the CAS API between the guest and the host. 

Once v0 is working I will test the upload perf to determine if we should also add a CAS API proxy.